### PR TITLE
REPLAY-1347 Improve image classification

### PR DESCRIPTION
**Pull request by maciejburda**
### What and why?

Improves image classification to get only contextual images into account.

### How?

It's accomplished by simplification of `ImageDataProvider` and moving some of it's responsibility to `UIImageViewRecorder` which enables reusability of this recorder.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

File: DatadogSessionReplay/Sources/Recorder/Utilities/ImageDataProvider.swift [hunk 1]
```diff
@@ -74,6 +60,36 @@ extension UIImage {
     }
 }
 
+fileprivate extension UIImage {
+    func compressToTargetSize(_ targetSize: Int) -> Data {
```
Comment (maciejburda) [inline DatadogSessionReplay/Sources/Recorder/Utilities/ImageDataProvider.swift:64]: improve

File: DatadogSessionReplay/Tests/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIImageViewWireframesBuilderTests.swift [hunk 2]
```diff
@@ -0,0 +1,94 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import XCTest
+@testable import DatadogSessionReplay
+
+class UIImageViewWireframesBuilderTests: XCTestCase {
```
Comment (maciejburda) [inline DatadogSessionReplay/Tests/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIImageViewWireframesBuilderTests.swift:10]: Coverage here could be increased. Maybe we could discuss the plan for testing more elaborate wireframes builders.

File: DatadogSessionReplay/Tests/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIImageViewWireframesBuilderTests.swift [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        if case let .imageWireframe(imageWireframe) = wireframes[1] {
+            XCTAssertEqual(imageWireframe.id, imageWireframeID)
+            XCTAssertEqual(imageWireframe.base64, "")
+        } else {
+            XCTFail("Second wireframe needs to be imageWireframe case")
+        }
+    }
+}
+
+class MockImageDataProvider: ImageDataProvider {
```
Comment (maciejburda) [inline DatadogSessionReplay/Tests/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIImageViewWireframesBuilderTests.swift:90]: we should probably move it somewhere more suitable

**Review by ncreated — CHANGES_REQUESTED** [review]
It looks OK 👌 , but causes regression compared to `develop` so we need to improve it. Some images result with base64 values that exceed the SDK core's max _event size_. In result we do not send replay.

On preventing such regression in the future - I think we need to consider adding integration tests of the same fashion as we have in RUM SDK (sending SR payload to HTTP server and asserting its delivery). Let's discuss it on daily.

File: DatadogSessionReplay/Tests/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIImageViewWireframesBuilderTests.swift (hunk 2 above)
Comment (maciejburda) [reply DatadogSessionReplay/Tests/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIImageViewWireframesBuilderTests.swift:10]: I like that. I'll keep these until we have snapshot in place. I'll describe the work in the ticket.

**Review by ncreated — APPROVED** [review]
Looks good 👌 💪

**Pull request merged**